### PR TITLE
#28 Bump spark commons version

### LINKED ISSUE #28: Bump absa spark commons version
## Description 
Bump absa spark commons version. Artefact ID changed as well. it contains the Spark version now. Need to come up with solution for that as well.

### PR DESCRIPTION
**Pull request by Zejnilovic**
Bump spark commons from 0.2 to 0.3.1

Had to make the commons a provided thing as it is spark dependent

Fixes #28

### CODE REVIEW

File: src/main/scala/za/co/absa/standardization/types/TypedStructField.scala [hunk 1]
```diff
@@ -28,11 +28,10 @@ import za.co.absa.standardization.time.DateTimePattern
 import za.co.absa.standardization.typeClasses.{DoubleLike, LongLike}
 import za.co.absa.standardization.types.parsers._
 import za.co.absa.standardization.validation.field._
-
 import scala.util.{Failure, Success, Try}
 
-sealed abstract class TypedStructField(structField: StructField)(implicit defaults: TypeDefaults)
-  extends StructFieldEnhancements(structField) with Serializable {
```
Comment (lsulak) [inline src/main/scala/za/co/absa/standardization/types/TypedStructField.scala:35]: there we go..such an ugly, hidden problem

**Review by lsulak — DISMISSED** [review]

File: src/main/scala/za/co/absa/standardization/types/TypedStructField.scala (hunk 1 above)
Comment (lsulak) [reply src/main/scala/za/co/absa/standardization/types/TypedStructField.scala:35]: btw, you can remove this from imports (I got "Unused import statement" on line 23)

**Comment by lsulak** [comment]
- [x]  code reviewed
- [x]  pulled
- [x]  built
- [x]  ran tests

**Review by lsulak — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -6,15 +6,27 @@
 - Dataframe in 
 - Standardized Dataframe out
 
-## Dependency
-SBT:
+## Usage
+
+### Needed Provided Dependencies
+
+Thi library needs following dependencies to be included in your project
```
Comment (jakipatryk) [inline README.md:13]: I think there is a typo ("The")

File: src/main/scala/za/co/absa/standardization/types/TypedStructField.scala (hunk 1 above)
Comment (benedeki) [reply src/main/scala/za/co/absa/standardization/types/TypedStructField.scala:35]: Sorry, this was my original mistake/implementation.

File: project/Dependencies.scala [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import sbt._
+
+object Dependencies {
+  private def getSparkVersionUpToMinor(sparkVersion: String): String =  {
+    val pattern = "([0-9]+)\\.([0-9]+)\\.([0-9]+)".r
+    val pattern(major, minor, patch) = sparkVersion
+    s"$major.$minor"
+  }
+
+  private def sparkFastTestsVersion(scalaVersion: String): String = if (scalaVersion.startsWith("2.12")) "1.1.0" else "0.23.0"
```
Comment (benedeki) [inline project/Dependencies.scala:26]: Wouldn't it be better to switch the logic? If starts with 2.11 0.23 else 1.1.0?

File: README.md [hunk 4]
```diff
@@ -6,15 +6,27 @@
...
+
+```sbt
+"org.apache.spark" %% "spark-core" % SPARK_VERSION,
+"org.apache.spark" %% "spark-sql" % SPARK_VERSION,
```
Comment (benedeki) [inline README.md:17]: Shouldn't these be probably provided?

File: project/Dependencies.scala [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  def getSparkVersion(scalaVersion: String): String = if (scalaVersion.startsWith("2.12")) "3.2.1" else "2.4.7"
+
+  def dependencyList(scalaVersion: String): Seq[ModuleID] = {
+    val sparkVersion = getSparkVersion(scalaVersion)
+    val sparkVersionUpToMinor = getSparkVersionUpToMinor(sparkVersion)
+    List(
+      "org.apache.spark" %% "spark-core" % sparkVersion % Provided,
+      "org.apache.spark" %% "spark-sql" % sparkVersion % Provided,
+      "za.co.absa" %% s"spark-commons-spark$sparkVersionUpToMinor" % "0.3.1" % Provided,
+      "za.co.absa" %% "spark-commons-test" % "0.3.1",
```
Comment (jakipatryk) [inline project/Dependencies.scala:37]: Is this one needed outside Test?

File: README.md (hunk 4 above)
Comment (Zejnilovic) [reply README.md:17]: My thinking was, that they are in `Needed Provided Dependencies` section. So you need to include them when you use this library. How you include them is not up to this readme to say.

File: project/Dependencies.scala (hunk 3 above)
Comment (Zejnilovic) [reply project/Dependencies.scala:26]: switched

**Review by jakipatryk — APPROVED** [review]
LGTM (just read the code)

**Review by lsulak — APPROVED** [review]

**Pull request merged**